subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/deps/ClassSetAnalysis.java

```diff
@@ -157,6 +157,9 @@ private DependentsSet findDirectDependents(String className) {
      * - the originating types of generated classes that need to be recompiled, since they wouldn't exist if the originating type is not reprocessed
      */
     public Set<String> getTypesToReprocess(Set<String> compiledClasses) {
+        if (compiledClasses.isEmpty()) {
```

Comment:
Regarding the first point of comment:
```
This includes:
- types which are annotated with aggregating annotations, as aggregating processors need to see them regardless of what has changed
```
Now I believe this is not true anymore if there is nothing to compile. Is there any aggregating processor out there that needs to run every time, even if there is nothing to compile? I doubt that makes sense, but I would just want to understand the change better

Reply:
"Regardless of what has changed" does not include "nothing has changed". I'm open for suggestion for better wording:) When nothing changed, processing makes no sense as you pointed out correctly.